src/blockchain/lending/liquidityPool.ts

```diff
@@ -0,0 +1,123 @@
+import { BaseProvider } from '@ethersproject/providers'
+import { parseEther, parseUnits } from '@ethersproject/units'
+import { Abi, ADDRESSES, LENDING_LIQUIDITY_POOL } from '@xtoken/abis'
+import { BigNumber, Contract } from 'ethers'
+
+import { getSignerAddress } from '../utils'
+
+import { getLiquidityPool } from './helper'
+
+// --- Liquidity Pool functions ---
+
+/**
+ * Supply USDC to Liquidity Pool
+ * @param amount amount of USDC without decimals
+ * @param provider
+ * @returns
+ */
+export const supply = async (amount: string, provider: BaseProvider) => {
+  const liquidityPool = await getLiquidityPool(provider)
+  const inputAmount = parseUnits(amount, 6)
+  const address = await getSignerAddress(provider)
+  const approvedAmount = await getApprovedAmountUSDC(address, provider)
+  if (approvedAmount.lt(inputAmount)) {
+    return Promise.reject(
+      new Error('Please approve the tokens before supplying')
```

Comment:
We can look at standardising the error message for approve tokens as well, similar to contract initialisation